src/View/Form/AbstractContext.php

```diff
@@ -0,0 +1,53 @@
+<?php
+namespace Cake\View\Form;
+
+use Cake\Network\Request;
+
+/**
+ * Provides common functionality for ContextInterface implementations.
+ */
+abstract class AbstractContext implements ContextInterface
+{
+    /**
+     * The request object.
+     *
+     * @var \Cake\Network\Request
+     */
+    protected $_request;
+
+    /**
+     * The request type of the attached form.
+     *
+     * @var string
+     */
+    protected $_requestType;
+
+    /**
+     * Constructor.
+     *
+     * @param \Cake\Network\Request $request The request object.
+     */
+    protected function __construct(Request $request)
+    {
+        $this->_request = $request;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setRequestType($requestType)
+    {
+        $this->_requestType = $requestType;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function val($field)
+    {
+        if ($this->_requestType === 'get') {
+            return $this->_request->query($field);
+        }
+        return $this->_request->data($field);
+    }
```

Comment:
I'm a huge fan of deduplication :)!
I agree that this class does not have a lot of code in it. However it might nonetheless help implementers of the ContextInterface. Especially if new methods get added to the interface we would be able to provide a default implementation in the abstract class.
Your previous suggestion with enhancing the constructor would also fit well into this abstract class...